pubtools/_quay/push_docker.py

```diff
@@ -507,6 +488,57 @@ def remove_old_signatures(
                     key,
                 )
 
+    def _fetch_digest(self, repo, tag, media_type):
+        """Fetch digest of repo and tag for given media type.
+
+        Args:
+            repo(str): repository name
+            tag(str): image tag
+            mtype(str): media type fo requested digest
+            (can be only application/vnd.docker.distribution.manifest.v2+json,
+            application/vnd.docker.distribution.manifest.v1+json)
+
+        Returns(str): manifest digest of the container
+        """
+        image_schema = "{host}/{namespace}/{repo}:{tag}"
+        dest_ref = image_schema.format(
+            host=self.quay_host,
+            namespace=self.target_settings["quay_namespace"],
+            repo=repo,
+            tag=tag,
+        )
+        v2s1_manifest = True if media_type == QuayClient.MANIFEST_V2S1_TYPE else False
+        digest = self.dest_quay_client.get_manifest_digest(dest_ref, v2s1_manifest=v2s1_manifest)
+        return digest
+
+    def fetch_missing_push_items_digests(self, push_items, target_settings):
+        """Fetch digests for media types which weren't originally pushed.
+
+        In order to be able to sign v1 for images which were pushed as
+        v2sch2 or to sign v2sch2 for images which were pushed as v1
+        fetch digests of those missing media types from quay and
+        set it to item metadata into  'new_digests' mapping
+
+        Args:
+            push_items(list): list of push items
+            target_settings(dict): target settings
+        """
+        for item in sorted(push_items):
+            item.metadata["new_digests"] = {}
+            missing_media_types = set(
+                [QuayClient.MANIFEST_V2S2_TYPE, QuayClient.MANIFEST_V2S1_TYPE]
+            ) - set(item.metadata["build"]["extra"]["image"]["media_types"])
+
+            # Always add v1sch1 due to possible digest change
```

Comment:
Also, this comment seems rather misleading and inaccurate. We want v2s1 because we have to sign it since customers may pull it. It's unrelated to digest change.